### ISSUE TITLE
UnphiMojo must optimize XMIR files too

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo#2729]
At the moment, the result XMIR's coming out of `UnphiMojo` contain `method` attributes. Let's make sure they don't.

**Comment by yegor256** [comment]
@maxonfjvipon please, help with this one, it's pretty urgent, it blocks https://github.com/objectionary/normalizer/issues/3

**Comment by yegor256** [comment]
@maxonfjvipon I believe, we should introduce a new unit test, which will take some `.eo` program, parse it to `.xmir` (using `ParseMojo` + `OptimizeMojo`) and then print the XMIR with the help of `XMIR` class from eo-parse. Then, it will take the XMI and transfer it toto `.phi` (using `PhiMojo`), then to `.xmir` (using `UnphiMojo`), and then print it with the help of the same XMIR class from eo-parse. The two printed EO programs must be the same, I believe.

**Comment by maxonfjvipon** [comment]
@yegor256 in most cases they will be different:
EO:
```
[x] > main
  x > y
```
XMIR:
```xml
<o name='main'>
  <o name='x'/>
  <o name='y' base='x' ref='...'/>
</o>
```
phi:
```
{main -> [x -> ∅, y -> ξ.x]}
```
unphi:
```xml
<o name='main'>
  <o name='x'/>
  <o base='$'/>
  <o name='y' base='x' method=''/>
</o>
```

**Comment by yegor256** [comment]
@maxonfjvipon OK, let's make this test:

  * EO to XMIR (via `ParseMojo` + `OptimizeMojo`)
  * XMIR to PHI (via `PhiMojo`)
  * PHI to XMIR (via `UnphiMojo`)
  * XMIR to EO (via [XMIR](https://github.com/objectionary/eo/blob/master/eo-parser/src/main/java/org/eolang/parser/XMIR.java))
  * assert that EO is parseable (via `ParseMojo`)

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#2737](https://github.com/objectionary/eo/issues/2737) is still not solved.

**Comment by 0pdd** [comment]
@yegor256 the only puzzle [#2737](https://github.com/objectionary/eo/issues/2737) is solved here.

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#2750](https://github.com/objectionary/eo/issues/2750) is still not solved; solved: [#2737](https://github.com/objectionary/eo/issues/2737).

**Comment by yegor256** [comment]
@rultor release, tag is `0.34.2`

**Comment by 0pdd** [comment]
@yegor256 2 puzzles [#2779](https://github.com/objectionary/eo/issues/2779), [#2780](https://github.com/objectionary/eo/issues/2780) are still not solved; solved: [#2737](https://github.com/objectionary/eo/issues/2737), [#2750](https://github.com/objectionary/eo/issues/2750).

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#2779](https://github.com/objectionary/eo/issues/2779) is still not solved; solved: [#2737](https://github.com/objectionary/eo/issues/2737), [#2750](https://github.com/objectionary/eo/issues/2750), [#2780](https://github.com/objectionary/eo/issues/2780).

**Comment by 0pdd** [comment]
@yegor256 all 4 puzzles are solved here: [#2737](https://github.com/objectionary/eo/issues/2737), [#2750](https://github.com/objectionary/eo/issues/2750), [#2779](https://github.com/objectionary/eo/issues/2779), [#2780](https://github.com/objectionary/eo/issues/2780).
